src/app/modules/shared/edit-password/edit-password.component.spec.ts

```diff
@@ -213,16 +220,28 @@ function setPasswords(oldPass: string, newPass: string, newPassConfirm: string)
   fixture.detectChanges();
 }
 
-function generateObservableResponse(messageCode: string, isSuccess: boolean): Observable<any> {
-  if (isSuccess) {
-    return new Observable((observer) => {
-      observer.next({ messageCode: messageCode });
-      observer.complete();
-    });
-  } else {
-    return new Observable((observer) => {
-      observer.error({ error: { messageCode: messageCode } });
-      observer.complete();
-    });
-  }
+function generateObservableResponse(arg: string | any, isSuccess: boolean): Observable<any> {
+  return isSuccess ? generateSuccessObservable(arg) : generateErrorObservable(arg);
+}
+
+function generateSuccessObservable(arg: string | any): Observable<any> {
+  return new Observable((observer) => {
+    observer.next(generateSuccessResponseValue(arg));
+    observer.complete();
+  });
+}
+
+function generateSuccessResponseValue(arg: string | any): any {
+  return typeof arg === 'string' ? { messageCode: arg } : arg;
+}
+
+function generateErrorObservable(arg: string | any): Observable<any> {
+  return new Observable((observer) => {
+    observer.error(generateErrorResponseValue(arg));
+    observer.complete();
+  });
+}
+
+function generateErrorResponseValue(arg: string | any): any {
+  return typeof arg === 'string' ? { error: { messageCode: arg } } : { error: arg };
```

Comment:
Reduce duplication? Do you need generateSuccessObservable() and generateErrorObservable(), or can do you it with a ternary in generateObservableResponse()?

```
function generateObservableResponse(arg: string | any, isSuccess: boolean): Observable<any> {
  return new Observable((observer) => {
    observer.next(isSuccess ? generateSuccessResponseValue(arg) : generateErrorResponseValue(arg));
    observer.complete();
  });
}
```